### ISSUE TITLE
Layers panel spruce-up

### ISSUE THREAD

**Issue by DesignSpaceBot** [adobe-photoshop/spaces-design#55]
1. Refactor the `LayerTree` component into `Layer` and `LayerFace` components. The former just consists of a single `LayerFace` component (for non-end-group layers) and a list of child `Layer` components. Hence, `Layer` is a true tree component. `LayerFace` contains everything that's needed to style each individual layer, and is the only draggable component.
2. Fix a drag-and-drop rendering bug where the drag target would flash back to its original position occasionally while dragging. There shouldn't be any rendering issues now.
3. Significantly improve styling, thanks to @timriot. That work isn't yet done, exactly, but it's now both closer to the designshop ideal, and well organized (using something like a BEM naming convention) for future work.

@volfied: since the changes to these components are so deep, it's probably easier just to read `Layer`, `LayerFace` and `PagesPanel`. I haven't made any major changes to the logic in any of these components, but I did smooth things over quite a bit. Happy to chat about anything that's unclear.

@chadrolfs: Please file bugs about any visual glitches you find with the layers panel once this is merged!

-- @iwehrman at [2014-10-16T23:41:10Z](https://github.com/adobe-photoshop/spaces-design-archive/pull/55)
